R/commonly_used_db_data.R

```diff
@@ -238,18 +240,27 @@ extract_vaccination_history <- function(con, touchstone_cov = "201710gavi", touc
 ##' @param country_ All countries if NULL. Or specify a vector of countries
 ##' @param year_ All years if NULL. Or specify a vector of years
 ##' @param age_ All age groups if NULL. Or specify a vector of age groups
+##' @param demographic_source one of demographic_source.code, this works for IU where demography is no a model run version
 ##' @export
 get_population <- function(con, touchstone_pop = "201710gavi-5", demographic_statistic = "int_pop", gender = "Both",
-                           country_ = NULL, year_ = NULL, age_ = NULL) {
-  if (!grepl("-", touchstone_pop)) {
-    touchstone_pop <- get_touchstone(con, touchstone_pop)
-    message("touchstone version is not specified. Lateset version is used.")
+                           country_ = NULL, year_ = NULL, age_ = NULL, demographic_source = NULL) {
+
+  if(is.null(demographic_source)){
+    if (!grepl("-", touchstone_pop)) {
+      touchstone_pop <- get_touchstone(con, touchstone_pop)
+      message("touchstone version is not specified. Lateset version is used.")
+    }
+    ## this table get population data as you wish
+    sql <- read_sql(system_file("sql/population.sql"))
+    pop_src <- touchstone_pop
+  } else {
+    sql <- read_sql(system_file("sql/population2.sql"))
```

Comment:
Minor point but you could update these to a more meaningful file name. Something like `touchstone_population.sql` and `demographic_population.sql` or something similar

Reply:
Done.